src/workspaces/features/RunPipelineDialog/DHIS2Widget.tsx

```
@@ -0,0 +1,310 @@
+              return item.name ?? `${item.level}`;
+            }
+          }
+
+          const foundItem = options.items.find((opt) => {
+            if (opt.__typename === "DHIS2OrganisationUnitLevel") {
+              return opt.level === item;
+            } else {
+              return opt.id === item;
+            }
+          });
+          if (foundItem?.__typename === "DHIS2OrganisationUnitLevel") {
+            return foundItem.name ?? `${foundItem.level}`;
+          } else {
+            return foundItem?.name ?? t("Unknown ID: {{id}}", { id: item });
+          }
+        })
+        .filter(Boolean);
+      return displayedNames.join(", ");
+    }
+
+    if (typeof value === "object" && value !== null) {
+      if (value.__typename === "DHIS2OrganisationUnitLevel") {
+        return value.name ?? `${value.level}`;
+      }
+
+      return (
+        value.name ??
+        (value.level
+          ? `${value.level}`
+          : t("Unknown ID: {{id}}", { id: value.id }))
+      );
+    }
+
+    const selectedItem = options.items.find((item) => {
+      if (item.__typename === "DHIS2OrganisationUnitLevel") {
+        return item.level === value;
+      } else {
+        return item.id === value;
+      }
+    });
+    if (selectedItem?.__typename === "DHIS2OrganisationUnitLevel") {
+      return selectedItem.name ?? `${selectedItem.level}`;
+    } else {
+      return selectedItem?.name ?? t("Unknown ID: {{id}}", { id: value });
+    }
+  };
+
+  useEffect(() => {
+    if (parameter.multiple && !Array.isArray(currentValue)) {
+      form.setFieldValue(parameter.code, []);
+    }
+  }, [form, parameter.multiple, parameter.code]);
+
+  const handleSelectionChange = useCallback(
+    (selectedValue: any) => {
+      if (parameter.multiple) {
+        const selectedIds = Array.isArray(selectedValue)
+          ? selectedValue.map((item) => item.id ?? item.level).filter(Boolean)
+          : [];
```

On reading, I feel like there is a logic issue : if the parameter is multiple and the selected value is not an array, _we ignore the selectedValue_ and return an empty array. Ignoring selection seems odd right ?

---

if the selection is not an array, signals there is an issue in the implementation and returning empty array seems fine, doesn't it?

---

If there is an unexpected behavior, it sounds wise to me to throw an exception (that will be triggered by Sentry) rather than failing silently (https://stackoverflow.com/a/77361)